### PR TITLE
testing vendor books and pages that they don't redirect

### PR DESCRIPTION
**Pull request by omehes**
added 2 test files to test book titles and pages that they don't redirect on vendor domain

### CODE REVIEW

File: fixtures/webview.py [hunk 1]
```diff
@@ -298,6 +298,23 @@ def rex_released_books(rex_base_url):
     return response.json()["books"]
 
 
+@pytest.fixture(params=gen_from_file(os.path.join(DATA_DIR, "openstax_books_uuids.txt")))
+def openstax_allbooks_uuids(request):
```
Comment (m1yag1) [inline fixtures/webview.py]: place a under bar between all and books. all_books

File: tests/webview/vendorbooks/test_vendorbooks_load.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    # GIVEN the openstax list
+    for op_book in home.featured_books.openstax_list:
+
+        books.append(op_book.title)
+
+    for book_title in books:
+
+        home = Home(selenium, vendor_base_url).open()
+
+        print("\nBOOK TITLE  : ", book_title)
```
Comment (m1yag1) [inline tests/webview/vendorbooks/test_vendorbooks_load.py]: remove print

File: tests/webview/vendorbooks/test_vendorbooks_load.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+        title_link = home.driver.find_element_by_link_text(book_title)
+        title_link.click()
+
+        current_url = home.current_url
+        print("CURRENT URL : ", current_url)
```
Comment (m1yag1) [inline tests/webview/vendorbooks/test_vendorbooks_load.py]: remove print

File: tests/webview/vendorbooks/test_vendorpages_load.py [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+        or openstax_allbooks_uuids == "16ab5b96-4598-45f9-993c-b8d78d82b0c6"
+        or openstax_allbooks_uuids == "bb62933e-f20a-4ffc-90aa-97b36c296c3e"
+    ):
+        # testing 3 Polish books as I cannot find appropriate indexes
+        home.driver.find_element_by_link_text("Rozwiązania zadań").click()
+
+        print("PAGE TITLE  : Rozwiązania zadań")
+
+        current_url = home.current_url
+        print("CURRENT URL : ", current_url)
```
Comment (m1yag1) [inline tests/webview/vendorbooks/test_vendorpages_load.py]: remove print

File: tests/webview/vendorbooks/test_vendorpages_load.py [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+    elif (
+        openstax_allbooks_uuids == "4eaa8f03-88a8-485a-a777-dd3602f6c13e"
+        or openstax_allbooks_uuids == "16ab5b96-4598-45f9-993c-b8d78d82b0c6"
+        or openstax_allbooks_uuids == "bb62933e-f20a-4ffc-90aa-97b36c296c3e"
+    ):
+        # testing 3 Polish books as I cannot find appropriate indexes
+        home.driver.find_element_by_link_text("Rozwiązania zadań").click()
+
+        print("PAGE TITLE  : Rozwiązania zadań")
```
Comment (m1yag1) [inline tests/webview/vendorbooks/test_vendorpages_load.py]: remove print

File: tests/webview/vendorbooks/test_vendorpages_load.py [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+
+    else:
+        # checking rest of the books that title and chapter/pages redirects
+
+        # WHEN a chapter is expanded and we navigate to one of its pages
+        chapter = toc.chapters[1]
+        chapter = chapter.click()
+        page = chapter.pages[1]
+        title = page.title
+        print("PAGE TITLE  : ", title)
```
Comment (m1yag1) [inline tests/webview/vendorbooks/test_vendorpages_load.py]: remove print

File: tests/webview/vendorbooks/test_vendorpages_load.py [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+        content = page.click()
+
+        # THEN we end up at the correct page
+        current_url = home.current_url
+        print("CURRENT URL : ", current_url)
```
Comment (m1yag1) [inline tests/webview/vendorbooks/test_vendorpages_load.py]: remove print

**Comment by omehes** [comment]
@m1yag1 changes done :)

File: tests/webview/vendorbooks/test_vendorbooks_load.py [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+    # GIVEN the openstax list
+    for op_book in home.featured_books.openstax_list:
+
+        books.append(op_book.title)
+
+    for book_title in books:
+
+        home = Home(selenium, vendor_base_url).open()
+
+        title_link = home.driver.find_element_by_link_text(book_title)
```
Comment (m1yag1) [inline tests/webview/vendorbooks/test_vendorbooks_load.py]: There is selenium driver code in the test... is there a reason this couldn't be in the page object?

File: tests/webview/vendorbooks/test_vendorbooks_load.py [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+@markers.vendor
+@markers.test_case("C593141")
+@markers.nondestructive
+def test_vendor_books_not_redirecting(vendor_base_url, selenium):
+
+    home = Home(selenium, vendor_base_url).open()
+    books = []
+
+    # GIVEN the openstax list
+    for op_book in home.featured_books.openstax_list:
```
Comment (m1yag1) [inline tests/webview/vendorbooks/test_vendorbooks_load.py]: Do you need to append all the book titles to the list? Couldn't you use a for loop and iterate over all the books and extract the title link from the page object.

```python
vendor_home = Home(selenium, vendor_base_url).open(()
for book in home.featured_books.openstax_list:
    title_link = book.title_link
    title_link.click()
    
    current_url = home.current_url
    data = requests.get(current_url)
    
    assert vendor_base_url in current_url
    assert 200 == data.status_code
```

Is there a reason something like ☝️ doesn't work?

File: tests/webview/vendorbooks/test_vendorpages_load.py [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+    home = Home(selenium, vendor_base_url).open()
+
+    content1 = Content(selenium, vendor_base_url, id=openstax_all_books_uuids).open()
+
+    content = content1.open()
+    content.header_nav.click_contents_button()
+    toc = content.table_of_contents
+
+    if openstax_all_books_uuids == "9d8df601-4f12-4ac1-8224-b450bf739e5f":
+        # testing American Government 2e book as it requires different indexing
```
Comment (m1yag1) [inline tests/webview/vendorbooks/test_vendorpages_load.py]: Comments should go above the line they are commenting about.
```
# testing American Government 2e book as it requires different indexing
if openstax_all_books_uuids == "9d8df601-4f12-4ac1-8224-b450bf739e5f":
```

File: tests/webview/vendorbooks/test_vendorpages_load.py [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+
+
+@markers.vendor
+@markers.test_case("C593142")
+@markers.nondestructive
+def test_vendor_pages_load(vendor_base_url, selenium, openstax_all_books_uuids):
+
+    home = Home(selenium, vendor_base_url).open()
+
+    content1 = Content(selenium, vendor_base_url, id=openstax_all_books_uuids).open()
```
Comment (m1yag1) [inline tests/webview/vendorbooks/test_vendorpages_load.py]: Need a better variable name than `content1`

File: tests/webview/vendorbooks/test_vendorbooks_load.py (hunk 9 above)
Comment (omehes) [reply tests/webview/vendorbooks/test_vendorbooks_load.py]: I tried this and other ways but the everpresent problem with this is the stale element. Karen and I tried various ways to overcome it. This is one of them. It reloads the base page after book is clicked and no stale element exception occurs.

File: tests/webview/vendorbooks/test_vendorbooks_load.py (hunk 8 above)
Comment (omehes) [reply tests/webview/vendorbooks/test_vendorbooks_load.py]: Similar reason as above. Stale element exception. I tried to replace it with book_cover_link function but getting stale element... very frustrating exception.

**Comment by omehes** [comment]
@m1yag1 I did the discussed changes: changed home.driver to selenium fixture and added note regarding stale element solution to the code

**Pull request merged**